tests/test_contextmanagers.py

```diff
@@ -0,0 +1,256 @@
+from __future__ import annotations
+
+import sys
+from collections.abc import AsyncGenerator, Generator
+from contextlib import (
+    AbstractContextManager,
+    AsyncExitStack,
+    ExitStack,
+    asynccontextmanager,
+    contextmanager,
+)
+
+if sys.version_info >= (3, 11):
+    from typing import Self
+else:
+    from typing_extensions import Self
+
+import pytest
+
+from anyio import AsyncContextManagerMixin, ContextManagerMixin
+
+pytestmark = pytest.mark.anyio
+
+
+class DummyContextManager(ContextManagerMixin):
+    def __init__(self, handle_exc: bool = False) -> None:
+        self.started = False
+        self.finished = False
+        self.handle_exc = handle_exc
+
+    @contextmanager
+    def __contextmanager__(self) -> Generator[Self]:
+        self.started = True
+        try:
+            yield self
+        except RuntimeError:
+            if not self.handle_exc:
+                raise
+
+        self.finished = True
+
+
+class DummyAsyncContextManager(AsyncContextManagerMixin):
```

Comment:
might be worth a test for a class that implements both AsyncContextManagerMixin and ContextManagerMixin

Reply:
I'm not sure why? The mix-in classes have no overlap whatsoever.

Reply:
Oh, I just realized: that's the problem. One could enter the CM synchronously AND asynchronously without first exiting.